addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -1444,6 +1444,19 @@ export class PosStore extends Reactive {
             }
         );
     }
+    async orderDetails(order) {
+        this.action.doAction("point_of_sale.action_view_pos_order", {
+            props: {
+                resId: order?.id,
+                onSave: (record) => {
+                    this.data.read("pos.order", [record.evalContext.id]);
+                    this.action.doAction({
+                        type: "ir.actions.act_window_close",
+                    });
+                },
+            },
+        });
+    }
```

Comment:
this would be about the same thing, only that you don't need to create an action 
```js
  this.dialog.add(
            FormViewDialog,
            {
                resModel: "pos.order",
                resId: this.getSelectedOrder().id,
            }

        );
    }
```